ClickHouse.Client/Formats/BinaryStreamReader.cs

```diff
@@ -44,7 +47,15 @@ public object Read(ArrayType arrayType)
             var data = arrayType.MakeArray(length);
             for (var i = 0; i < length; i++)
             {
-                data.SetValue(ClearDBNull(Read(arrayType.UnderlyingType)), i);
+                var val = ClearDBNull(Read(arrayType.UnderlyingType));
+                if (arrayType.UnderlyingType.TypeCode == ClickHouseTypeCode.Nested)
+                {
+                    //data.SetValue(val, i);
```

Comment:
Insertion fails here

Reply:
The NestedType appeared to have overwritten the propery FrameworkType => base.FrameworkType.MakeArrayType();
causing a double array. I'm not sure why this is, but added a check if the type is already of an Array, and only when not convert it to an Array type.
All unit tests pass now.